transports/janus_websockets.c

```diff
@@ -790,17 +794,46 @@ static int janus_websockets_callback(struct libwebsocket_context *this,
 			}
 			if(!ws_client->destroy && !g_atomic_int_get(&stopping)) {
 				janus_mutex_lock(&ws_client->mutex);
-				/* Shoot all the pending messages first */
+				/* Check if we have a pending/partial write to complete first */
+				if(ws_client->partial_message && ws_client->partial_offset > 0
+						&& !ws_client->destroy && !g_atomic_int_get(&stopping)) {
+					int pending = strlen(ws_client->partial_message) - ws_client->partial_offset;
+					JANUS_LOG(LOG_VERB, "[WSS-%p] Completing pending WebSocket write (still need to write last %d/%zu bytes)...\n",
+						wsi, pending, strlen(ws_client->partial_message));
+					int sent = libwebsocket_write(wsi, (unsigned char *)(ws_client->partial_message + ws_client->partial_offset), pending, LWS_WRITE_TEXT);
```

Comment:
> Now that I think about this interface more, and see that the example does not worry about the tail part of the framing (which goes in the POST_PADDING space) landing in the middle of the message if the write is partial, I think I see how the interface works - multiple write calls can continue the same websocket framed message (it is over tcp after all), the websocket client state must remember if the previous write was incomplete (and if so, omit the head part of the framing).

Yes, that's what I thought: not sure if it checks this by looking at the beginning of the new write to match it to what it couldn't write before, even though that sounds feeble. I guess I might add a shared buffer for all writes, but if we make it static we're putting a cap on the max size of packets (and possibly reserving much more than needed for the whole duration of the session), and if we make it dynamic we'll have to realloc when needed.